h2o-algos/src/main/java/hex/gam/GAMModel.java

```diff
@@ -286,6 +287,13 @@ public long progressUnits() {
 
     public long _seed = -1;
 
+    @Override
+    protected void collectAllFrames(Map<String, Frame> map) {
+      super.collectAllFrames(map);
+      collectFrame(map, "plug_values", _plug_values);
```

Comment:
interesting idea, AFAIK reflection does not contain information about generics, we could find all Key typed params but will not be able to find those that are Key<Frame>

Reply:
Or force a method on an algo.

Reply:
@honzasterba see 
https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/lang/reflect/Field.html#getGenericType()
https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/lang/reflect/Method.html#getGenericParameterTypes()
https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/lang/reflect/ParameterizedType.html

all this is available since 1.5 and H2O-3 is actually using those in `ReflectionUtils`

Reply:
interesting, but this still does not help for the case of DeepLearningModel.initial_weights which is Key[], we could just take all Key and Key[] params and do DKV.get() and look at the object type and decide based on that

Reply:
agree, this works only for code, there's still type erasure for instances of course, so you can't obtain any magic information from objects stored in DKV

Reply:
after reflection (pun intended), for cases like `Key[]` we could consider that they should be declared as `Key<Frame>[]` arrays, and then reflection would work...

Anyway, as much as I also like @mn-mikke suggestion, it may require too many changes, idk..